src/rdkafka_offset.c

```diff
@@ -715,7 +718,7 @@ rd_kafka_offsets_store (rd_kafka_t *rk,
                 ok_cnt++;
         }
 
-        return offsets->cnt > 0 && ok_cnt < offsets->cnt ?
+        return offsets->cnt > 0 && ok_cnt == 0 ?
```

Comment:
The alternative is to remove this check altogether. I think I agree what you have is the best option.

Reply:
This is sort of an optimization on the caller/user's behalf: if all partitions failed the user won't have to check the error code for each individual partition.
Questionable value, but it is there now.